### PR TITLE
paperless-ngx: 1.14.4 -> 1.16.1, python3Packages.python-ipware: init at 0.9.0,  python3Packages.tika-client: init at 0.1.0

### PR DESCRIPTION
**Pull request by e1mo**
###### Description of changes

Various smaller new features (quick filters from document detail, explanations for relative dates, pagination for tasks) and various other bug fixes and minor enhancements. Especially noteworthy for v1.14.5 seems a security fix around the display of texts in the frontend[^1].

Full changelog available at
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.14.5
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.15.0
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.15.1
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.16.0
  https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.16.1

[^1]: Since the [bug report](https://huntr.dev/bounties/94517f3f-ed86-4d88-bce1-6e9ba11fe1c2/) is not yet public, it's not possible to judge the severity of it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by leona-ya — APPROVED** [review]
looks good over all, also tested on my instance.

there's one small typo/consistency in a commit message: `Init` vs `init`

**Comment by joelkoen** [comment]
v1.15.1 was just released

**Comment by e1mo** [comment]
Thanks @joelkoen. I've updated the PR to paperless-1.15.1 and also fixed the capitalization typo bought mentioned by @leona-ya.

**Review by leona-ya — APPROVED** [review]
Tested again, looks good.

I still suggest backporting the PR since it fixes the security bug, and 1.15.0 and 1.15.1 have no breaking changes as far as I can see

**Review by joelkoen — APPROVED** [review]
Tested on my instance too

File: pkgs/development/python-modules/python-ipware/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/python-ipware/default.nix:18]:
```suggestion
  pythonImportsCheck = [
    "ipware"
  ];

  nativeCheckInputs = [
    pytestCheckHook
  ];
```

Please enable tests like so.

File: pkgs/development/python-modules/python-ipware/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/python-ipware/default.nix:23]:
```suggestion
```

File: pkgs/development/python-modules/python-ipware/default.nix
Comment (e1mo) [reply pkgs/development/python-modules/python-ipware/default.nix:20]: Thanks for catching my oversight! I had to switch to using the source from GitHub directly because the unittests are not included in the PyPi tarball.

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Would really like one of the maintainers to approve.

**Comment by leona-ya** [comment]
Would (at least after that is merged) suggest me as another maintainer. I know this is not helpful now for this issue, but maybe for the future.

**Review by lukegb — APPROVED** [review]

File: pkgs/development/python-modules/tika-client/default.nix
Comment (leona-ya) [inline pkgs/development/python-modules/tika-client/default.nix:33]: Hint: This magic is not the required one for the tests.

**Comment by leona-ya** [comment]
Paperless also works with the PR, so if we decide that fixing the test is too much effort, it's okay for me.

File: pkgs/development/python-modules/tika-client/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/tika-client/default.nix:33]: We require explicit test setups, since the implicit one uses `python setup.py test`, which is deprecated.

Comment (e1mo) [reply pkgs/development/python-modules/tika-client/default.nix:33]: Without magic, the tests failed for me due to missing imports. @mweinelt I'm a bit unsure on how to understand your remark. Should I change anything here or is it fine the way it is?

**Comment by e1mo** [comment]
> so if we decide that fixing the test is too much effort

As context to why I personally see this as to much effort: I do think that fixing the tests would require to much effort since tika is not packaged or even modularized in nixpkgs yet.

File: pkgs/development/python-modules/tika-client/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/tika-client/default.nix:33]: > Hint: This magic is not the required one for the tests.

The correct package would be `python-magic`.

> I'm a bit unsure on how to understand your remark. Should I change anything here or is it fine the way it is?

Without a check hook the check phase runs `python setup.py test`, which is deprecated. So yes, I want you to set up `pytestCheckHook` to a reasonable degree, or disable the tests entirely by setting `doCheck = false; # <reason>`. Leaving it hanging like that is not acceptable.

Comment (e1mo) [reply pkgs/development/python-modules/tika-client/default.nix]: Thanks for elaborating. Due to the amount of failing (to their dependency on network / external services) I went with your proposal of just disabling all tests with a note on why.

File: pkgs/development/python-modules/tika-client/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/tika-client/default.nix:21]: No `pythonPackages` usage within `python-modules`, please. Please list all dependencies as function arguments.

```suggestion
  propagatedBuildInputs = [
```

Comment (e1mo) [reply pkgs/development/python-modules/tika-client/default.nix]: Thanks, fixed it.

**Pull request merged**
